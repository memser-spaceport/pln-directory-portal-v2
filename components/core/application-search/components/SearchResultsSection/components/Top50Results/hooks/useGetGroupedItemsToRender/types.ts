import { ForumFoundItem, FoundItem } from '@/services/search/types';

export type State = Record<
  string,
  {
    items: (FoundItem | ForumFoundItem)[];
    hasMore: boolean;
    showAll: boolean;
  }
>;

export type Action =
  | {
      type: 'setInitialState';
      payload: State;
    }
  | {
      type: 'setShowAll';
      payload: {
        groupName: string;
        showAll: boolean;
      };
    };
