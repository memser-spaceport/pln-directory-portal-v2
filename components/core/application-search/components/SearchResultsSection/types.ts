import { ForumFoundItem, FoundItem } from '@/services/search/types';

export type AllFoundItems = FoundItem[] | ForumFoundItem[];
