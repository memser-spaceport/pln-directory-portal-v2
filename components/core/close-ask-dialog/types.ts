import { ITeam } from '@/types/teams.types';

export enum AskCloseReasons {
  FULLY_ADDRESSED = 'Fully Addressed',
  PARTIALLY_ADDRESSED = 'Partially Addressed',
  NO_LONGER_NEEDED = 'No Longer Needed',
  UNADRESSABLE = 'Unadressable',
  DUPLICATE = 'Duplicate/Created in Error',
  OTHER = 'Other',
}

export type CloseAskForm = {
  reason: string;
  resolvedBy: MemberOption | null;
  comments?: string;
};

export interface CloseAskDialogProps {
  team: ITeam;
  data: {
    title: string;
    description: string;
    tags: string[];
    uid: string;
    teamUid: string;
  };
  onClose: () => void;
  isVisible: boolean;
  onSuccess: () => Promise<void>;
}

export type MemberOption = {
  isVerified: boolean;
  name: string;
  profile: string;
  uid: string;
};
