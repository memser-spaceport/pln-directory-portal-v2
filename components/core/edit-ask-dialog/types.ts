export type EditAskForm = {
  title: string;
  description: string;
  tags: string[];
  disabled?: boolean;
};
