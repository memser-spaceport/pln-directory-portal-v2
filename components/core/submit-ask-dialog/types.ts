export type SubmitAskForm = {
  title: string;
  description: string;
  tags: string[];
};
