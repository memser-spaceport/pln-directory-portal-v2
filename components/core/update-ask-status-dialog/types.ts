import { ITeam } from '@/types/teams.types';

export enum AskCloseReasons {
  ACTIVE = 'Active',
  FULLY_ADDRESSED = 'Fully Addressed',
  PARTIALLY_ADDRESSED = 'Partially Addressed',
  NO_LONGER_NEEDED = 'No Longer Needed',
  UNADRESSABLE = 'Unaddressable',
  DUPLICATE = 'Duplicate/Created in Error',
  OTHER = 'Other',
}

export type CloseAskForm = {
  reason: string;
  resolvedBy: MemberOption | null;
  comments?: string;
  disabled?: boolean;
};

export interface CloseAskDialogProps {
  team: ITeam;
  data: {
    title: string;
    description: string;
    tags: string[];
    uid: string;
    teamUid: string;
  };
  onClose: () => void;
  isVisible: boolean;
  onSuccess: () => Promise<void>;
}

export type MemberOption = {
  isVerified: boolean;
  name: string;
  profile: string;
  uid: string;
};
