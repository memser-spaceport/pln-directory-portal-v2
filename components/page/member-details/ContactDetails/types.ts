export interface TEditContactForm {
  officeHours: string | null;
  telegram: string | null;
  github: string | null;
  linkedin: string | null;
  discord: string | null;
  twitter: string | null;
  email: string | null;
}
