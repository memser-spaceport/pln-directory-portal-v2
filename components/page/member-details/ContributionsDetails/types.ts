export interface TEditContributionsForm {
  name: Record<string, string>;
  role: string;
  description: string;
  startDate: string | null;
  endDate: string | null;
}
