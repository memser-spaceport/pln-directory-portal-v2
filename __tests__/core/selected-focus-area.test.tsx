import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';
import SelectedFocusAreas from '@/components/core/selected-focus-area';
import { IFocusArea } from '@/types/shared.types';

// Mock FocusAreaDisplay to isolate SelectedFocusAreas tests
jest.mock('@/components/core/focus-area-display', () => ({
  __esModule: true,
  default: ({ rawData, selectedItems }: any) => (
    <div data-testid="mock-focus-area-display">
      {`rawData:${rawData.length},selectedItems:${selectedItems.length}`}
    </div>
  ),
}));

describe('SelectedFocusAreas', () => {
  const baseFocusAreas: any = [
    { uid: '1', title: 'Area 1', children: [] },
    { uid: '2', title: 'Area 2', children: [] },
  ];

  it('renders the section title', () => {
    render(
      <SelectedFocusAreas focusAreas={baseFocusAreas} selectedFocusAreas={[]} />
    );
    expect(screen.getByText('Focus Area')).toBeInTheDocument();
  });

  it('renders FocusAreaDisplay with correct props when nothing is selected', () => {
    render(
      <SelectedFocusAreas focusAreas={baseFocusAreas} selectedFocusAreas={[]} />
    );
    const display = screen.getByTestId('mock-focus-area-display');
    expect(display).toHaveTextContent('rawData:2,selectedItems:0');
  });

  it('renders FocusAreaDisplay with correct props when some are selected', () => {
    render(
      <SelectedFocusAreas focusAreas={baseFocusAreas} selectedFocusAreas={[baseFocusAreas[0]]} />
    );
    const display = screen.getByTestId('mock-focus-area-display');
    expect(display).toHaveTextContent('rawData:2,selectedItems:1');
  });

  it('renders FocusAreaDisplay with empty focusAreas', () => {
    render(
      <SelectedFocusAreas focusAreas={[]} selectedFocusAreas={[]} />
    );
    const display = screen.getByTestId('mock-focus-area-display');
    expect(display).toHaveTextContent('rawData:0,selectedItems:0');
  });

  it('matches snapshot for regression safety', () => {
    const { asFragment } = render(
      <SelectedFocusAreas focusAreas={baseFocusAreas} selectedFocusAreas={[baseFocusAreas[1]]} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  // Edge case: selectedFocusAreas contains an item not in focusAreas
  it('handles selectedFocusAreas not present in focusAreas', () => {
    const rogue = { uid: '999', title: 'Rogue', children: [] };
    render(
      <SelectedFocusAreas focusAreas={baseFocusAreas} selectedFocusAreas={[rogue]} />
    );
    const display = screen.getByTestId('mock-focus-area-display');
    expect(display).toHaveTextContent('rawData:2,selectedItems:1');
  });
}); 