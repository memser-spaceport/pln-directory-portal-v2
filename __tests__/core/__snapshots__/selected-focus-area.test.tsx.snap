// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedFocusAreas matches snapshot for regression safety 1`] = `
<DocumentFragment>
  <div
    class="jsx-2bcc6b9659845454 focus"
  >
    <h2
      class="jsx-2bcc6b9659845454 focus__title"
    >
      Focus Area
    </h2>
    <div
      data-testid="mock-focus-area-display"
    >
      rawData:2,selectedItems:1
    </div>
  </div>
</DocumentFragment>
`;
